crates/cheatcodes/spec/src/vm.rs

```diff
@@ -260,6 +260,23 @@ interface Vm {
         uint64 depth;
     }
 
+    /// The result of the `stopAndReturnDebugTraceRecording` call
+    struct DebugStep {
+        /// The stack on the step of the run.
+        uint256[] stack;
+        /// The memory data on the step of the run.
+        uint8[] memoryData;
+        /// The opcode that was accessed.
+        uint8 opcode;
+        /// The call depth of the step.
+        uint64 depth;
+        /// The instruction result.
+        /// see: https://github.com/bluealloy/revm/blob/5a47ae0d2bb0909cc70d1b8ae2b6fc721ab1ca7d/crates/interpreter/src/instruction_result.rs#L6-L50.
+        uint8 instructionResult;
```

Comment:
this is not something thats stable and should not be included, at most it should be a separate enum thats like { stop, return, revert, panic/error }

Reply:
do you mean by copy a enum of the whole `InstructionResult`: https://github.com/bluealloy/revm/blob/5a47ae0d2bb0909cc70d1b8ae2b6fc721ab1ca7d/crates/interpreter/src/instruction_result.rs#L6-L50

Or you mean to have a more context specific enum defined here?

for context, I actually just need a way to detect "out of gas" and this was the only one I figured out. Like to know if a customized enum is preferred, what would be expected to be put.

Reply:
okay, change this to a `isOutOfGas` flag instead to avoid expose internal instruction result details.